labelbox/client.py

```diff
@@ -709,3 +707,189 @@ def get_data_row_ids_for_external_ids(
+                                   ['ontologies', 'nextCursor'])
+
+    def get_feature_schema(self, feature_schema_id):
+        """
+        Fetches a feature schema. Only supports top level feature schemas.
+
+        Args:
+            feature_schema_id (str): The id of the feature schema to query for
+        Returns:
+            FeatureSchema
+        """
+
+        query_str = """query rootSchemaNodePyApi($rootSchemaNodeWhere: RootSchemaNodeWhere!){
+              rootSchemaNode(where: $rootSchemaNodeWhere){%s}
+        }""" % query.results_query_part(Entity.FeatureSchema)
+        res = self.execute(
+            query_str,
+            {'rootSchemaNodeWhere': {
+                'featureSchemaId': feature_schema_id
+            }})['rootSchemaNode']
+        res['id'] = res['normalized']['featureSchemaId']
+        return Entity.FeatureSchema(self, res)
+
+    def get_feature_schemas(self, name_contains):
+        """
+        Fetches top level feature schemas with names that match the `name_contains` string
+
+        Args:
+            name_contains (str): the string to search top level feature schema names by
+        Returns:
+            PaginatedCollection of FeatureSchemas with names that match `name_contains`
+        """
+        query_str = """query rootSchemaNodesPyApi($search: String, $filter: RootSchemaNodeFilter, $from : String, $first: PageSize){
+            rootSchemaNodes(where: {filter: $filter, search: $search}, after: $from, first: $first){
+                nodes {%s}
+                nextCursor
+            }
+        }
+        """ % query.results_query_part(Entity.FeatureSchema)
+        params = {'search': name_contains, 'filter': {'status': 'ALL'}}
+
+        def rootSchemaPayloadToFeatureSchema(client, payload):
+            # Technically we are querying for a Schema Node.
+            # But the features are the same so we just grab the feature schema id
+            payload['id'] = payload['normalized']['featureSchemaId']
+            return Entity.FeatureSchema(client, payload)
+
+        return PaginatedCollection(self, query_str, params,
+                                   ['rootSchemaNodes', 'nodes'],
+                                   rootSchemaPayloadToFeatureSchema,
+                                   ['rootSchemaNodes', 'nextCursor'])
+
+    def create_ontology_from_feature_schemas(self, name, feature_schema_ids):
+        """
+        Creates an ontology from a list of feature schema ids
+
+        Args:
+            name (str): Name of the ontology
+            feature_schema_ids (List[str]): List of feature schema ids corresponding to
+                top level tools and classifications to include in the ontology
+        Returns:
+            The created Ontology
+        """
+        tools, classifications = [], []
+        for feature_schema_id in feature_schema_ids:
+            feature_schema = self.get_feature_schema(feature_schema_id)
+            tool = feature_schema.normalized['tool']
```

Comment:
I'm not good at python so i can't tell if this handles it, but `tool` will only be defined on a tool kind schema, for a classification the field is `type` (i know, its dumb)

Reply:
Nice catch!

Reply:
Fixed

Reply:
thanks!